parquet-hadoop/src/test/java/org/apache/parquet/hadoop/TestParquetFileWriter.java

```diff
@@ -286,6 +286,9 @@ public void testWriteRead() throws Exception {
     w.endColumn();
     w.endBlock();
     w.end(new HashMap<String, String>());
+    // Although writer is already closed in previous end(),
+    // explicitly close it again to verify double close behavior.
+    w.close();
```

Comment:
Should we give a best practice in the test, and turn this into a try-with-resource? The `ParquetFileWriter` is auto-closable.

Reply:
I didn't do this for two reasons:

1 The verification codes following this write operation use multiple local variables declared during the write (e.g. c1Starts c1p1Starts c1Ends). It's a pain to pre declare all these variables before the try block.

2 Each test in this test file is coded like this, so it's a pain to refactor all usages of ParquetFileWriter here.

Reply:
Maybe we can just turn this one into a try-with-resource style.

Reply:
Ah yes, let's keep it like this for now.